js/source/source.js

```diff
@@ -32,7 +33,22 @@ exports._loadTileJSON = function(options) {
             redoPlacement: this._redoTilePlacement ? this._redoTilePlacement.bind(this) : undefined
         });
 
-        this.fire('load');
+        // if index is defined, fetch the index json, then extend the pyramid
+        if (tileJSON.index) {
```

Comment:
This is the entry point for the difference in behavior. If an index exists (as a url) on the data source "tileJSON" the index gets requested. The returned index is added to the tile pyramid.